### PR TITLE
[INLONG-10006][Agent] Add MongoDB data source for Agent

### LINKED ISSUE #10006: [Feature][Agent] Add MongoDB data source for Agent
### Description

Add MongoDB data source for Agent

### Use case

Add MongoDB data source for Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haifxu**
### Prepare a Pull Request

- Fixes #10006 

### Motivation

Add MongoDB data source for Agent

### Modifications

1. Add MongoDB Task and Instance.
2. Start a new thread, let `Debezium` collect MongoDB data, and then put it into a queue(named `debeziumQueue`). The agent will regularly read messages from this queue.
3. The offset information file generated by Debezium is placed in the `/inlong-agent` directory.

### Verifying this change

image

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/MongoDBSource.java [hunk 1]
```diff
@@ -17,56 +17,154 @@
...
     protected List<SourceData> readFromSource() {
-        return null;
-    }
-
-    @Override
-    public Message read() {
-        return null;
+        List<SourceData> dataList = new ArrayList<>();
+        try {
+            SourceData sourceData = debeziumQueue.poll(1, TimeUnit.SECONDS);
```
Comment (justinwwhuang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/MongoDBSource.java:159]: You can read multiple entries at once here, just limit the total length

Comment (haifxu) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/MongoDBSource.java]: Done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/MongoDBSource.java [hunk 2]
```diff
@@ -17,56 +17,160 @@
...
+            while (size < BATCH_READ_LINE_TOTAL_LEN) {
+                SourceData sourceData = debeziumQueue.poll(1, TimeUnit.SECONDS);
+                if (sourceData != null) {
+                    size += sourceData.getData().length;
+                    dataList.add(sourceData);
+                } else {
+                    break;
+                }
+            }
+            LOGGER.info("read {} messages from debezium queue", dataList.size());
```
Comment (justinwwhuang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/MongoDBSource.java:169]: log level should be debug here

Comment (haifxu) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/MongoDBSource.java]: Done

**Review by justinwwhuang — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**